### PR TITLE
Vault test matrix

### PR DESCRIPTION
**Pull request by dadgar**
This PR introduces an E2E test to ensure that Nomad works with a matrix of Vault versions

### CODE REVIEW

File: e2e/vault/vault_test.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+}
+
+// TestVaultCompatibility tests compatibility across Vault versions
+func TestVaultCompatibility(t *testing.T) {
+	h := newHarness(t)
+	vaultBinaries := h.reconcile()
+
+	for version, vaultBin := range vaultBinaries {
+		t.Run(version, func(t *testing.T) {
+			vbin := vaultBin
```
Comment (schmichael) [inline e2e/vault/vault_test.go]: Doesn't this have to be the first line in the for loop -- outside the func?

File: vendor/vendor.json [hunk 2]
```diff
@@ -195,7 +195,7 @@
...
 		{"path":"github.com/hashicorp/serf/serf","checksumSHA1":"QrT+nzyXsD/MmhTjjhcPdnALZ1I=","revision":"80ab48778deee28e4ea2dc4ef1ebb2c5f4063996","revisionTime":"2018-05-07T23:19:28Z"},
-		{"path":"github.com/hashicorp/vault/api","checksumSHA1":"+B4wuJNerIUKNAVzld7CmMaNW5A=","revision":"8575f8fedcf8f5a6eb2b4701cb527b99574b5286","revisionTime":"2018-09-06T17:45:45Z"},
+		{"path":"github.com/hashicorp/vault/api","checksumSHA1":"DP7dd8OErZVF0q+XfPo0RGkDcLk=","revision":"6e8d91a59c34bd9f323397c30be9651422295c65","revisionTime":"2018-09-19T17:09:49Z"},
 		{"path":"github.com/hashicorp/vault/helper/compressutil","checksumSHA1":"bSdPFOHaTwEvM4PIvn0PZfn75jM=","revision":"8575f8fedcf8f5a6eb2b4701cb527b99574b5286","revisionTime":"2018-09-06T17:45:45Z"},
 		{"path":"github.com/hashicorp/vault/helper/consts","checksumSHA1":"QNGGvSYtwk6VCkj4laZPjM2301E=","revision":"8575f8fedcf8f5a6eb2b4701cb527b99574b5286","revisionTime":"2018-09-06T17:45:45Z"},
 		{"path":"github.com/hashicorp/vault/helper/hclutil","checksumSHA1":"RlqPBLOexQ0jj6jomhiompWKaUg=","revision":"8575f8fedcf8f5a6eb2b4701cb527b99574b5286","revisionTime":"2018-09-06T17:45:45Z"},
```
Comment (schmichael) [inline vendor/vendor.json]: Should we keep all of the vault packages at the same revision?

Comment (dadgar) [reply vendor/vendor.json]: Not really, the other packages didn't have relevant changes to pull in.

File: e2e/vault/vault_test.go [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+	}
+
+	// Create the directory
+	if err := os.Mkdir(h.binDir, 0700); err != nil {
+		h.t.Fatalf("failed to make directory: %v", err)
+	}
+	if err := os.Chmod(h.binDir, 0700); err != nil {
+		h.t.Fatalf("failed to chmod: %v", err)
+	}
+}
```
[the remark this answers is not in the document]
Comment (dadgar) [reply e2e/vault/vault_test.go]: MkdirAll is useful for creating sub directories that don't exist. In this case the parent exists so we are just creating the subdirectory directly so mkdir is fine. The second chmod is b/c it wasn't properly chmoding unless I added the second call 🤷‍♂️

File: e2e/vault/vault_test.go (hunk 1 above)
Comment (dadgar) [reply e2e/vault/vault_test.go]: Yep good catch!

**Pull request merged**